libraries/AP_NavEKF2/AP_NavEKF2_MagFusion.cpp

```diff
@@ -390,6 +391,17 @@ void NavEKF2_core::FuseMagnetometer()
         MagPred[1] = DCM[1][0]*magN + DCM[1][1]*magE  + DCM[1][2]*magD + magYbias;
         MagPred[2] = DCM[2][0]*magN + DCM[2][1]*magE  + DCM[2][2]*magD + magZbias;
 
+        if (pending_z_mag_correction && frontend->_mag_learn_limit > 0) {
+            pending_z_mag_correction = false;
+            float correction = magDataDelayed.mag[2] - MagPred[2];
```

Comment:
I can see that *.mag and MagPred are defined as 3-element vectors.  Do we assume that indices 0, 1, and 2 are X, Y, and Z, respectively?  Should these be #defined, const, or constexpr so that it’s more clear??  E.g. MagPred[Z_AXIS]

Reply:
they are X, Y and Z axis, yes, used throughout the code base